storage/gcloud/aio/storage/storage.py

```diff
@@ -591,12 +591,14 @@ async def _download(
         # aiohttp and requests automatically decompress the body if this
         # argument is not passed. We assume that if the Accept-Encoding header
         # is present, then the client will handle the decompression
-        auto_decompress = 'accept-encoding' not in {k.lower() for k in headers}
+        auto_decompress = None  # inherit the Session default setting
+        if 'accept-encoding' not in {k.lower() for k in headers}:
```

Comment:
```suggestion
        if 'accept-encoding' in {k.lower() for k in headers}:
```

Reply:
I believe we want want Kevin has here. If `'accept-encoding' in {k.lower() for k in headers}` is `True`, that means that the session passes `auto_decompress` value implicitly in `session`, and we want `None`. We only want to set `auto_decompress` if `'accept-encoding' in {k.lower() for k in headers}` is False, that is `'accept-encoding'` is not in the headers.